.travis.yml

```diff
@@ -35,8 +35,16 @@
 language: python
 python:
   - 2.7
+# The env section allows us to specify multiple, different environments
+# in which the test suite must be run. The environment variables set
+# here are picked up in the bin/install_optional.sh script.
+env:
+  matrix:
+    - WITH_TERMCOLOR=0
```

Comment:
This is nice, although does it mean that all tests get run twice, or just the ones which depend on WITH_TERMCOLOR?

Reply:
It means that all tests get run twice. I think this is necessary as I wouldn't want to rely on people cherry-picking which tests to run for a given environment. In theory we could pick-up the value of WITH_TERMCOLOR in the code that sets-up the pytest fixtures. However, the tests are written in such a way that they are valid with or without termcolor being installed - there's no explicit reference to that package within them.